File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.cpp
class Solution {
public:
    static bool comparator(pair<int, char> p1, pair<int, char> p2){
        if(p1.first > p2.first) return true;
        if(p1.first < p2.first) return false;
        return p1.second < p2.second;
    }
    string frequencySort(string s) {
        pair<int, char> freq[123];
        for(int i = 0; i < 123; i++){
            freq[i].first = 0;
            freq[i].second = '\0' + i;
        }
        for(int  i = 0; i < s.length(); i++){
            freq[s[i] - '\0'].first++;
        }
        sort(freq, freq + 123, comparator);
        s = "";
        for (int i = 0; i < 123; i++) {
          for(int j = 0; j < freq[i].first; j++) {
             s += freq[i].second;
         }
       }
       return s;
    }
};

File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.py
class Solution:

    def frequencySort(self,s:str) -> str:
        freq={}
        for char in s:
            freq[char]=freq.get(char,0)+1

        sorted_data = dict(sorted(freq.items(), key=lambda x: x[1],reverse=True))
        a=""
        for elem in sorted_data:
            
            a+=sorted_data[elem]*elem

        return a